users/token/token.go

```diff
@@ -0,0 +1,113 @@
+//Package token provides password recovery token generation with TTL
+//Token is sent by email to user as part of recovery URL
+//Token contains 32 bytes where first 4 bytes are exparation time and 28 bytes random string
+// exparation-email signed by secret signature
+package token
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mainflux/mainflux/users/mail"
+	"golang.org/x/crypto/bcrypt"
+)
+
+var (
+	emailLength   = 254
+	ttlLength     = 4
+	secretLength  = 20
+	tokenLength   = ttlLength + emailLength // Max token length is 4 bytes + max email length
+	hashCost      = 10
+	tokenDuration = 5                      // Recovery token TTL in minutes, reperesents token time to live
+	secret        = "fcERNb7KpM3WyAmguJMZ" // Random string for secret key, required for signing
```

Comment:
You should probably move this to the env var if you haven't already. Also, I recommend that you use https://github.com/dgrijalva/jwt-go here.

Reply:
IMHO JWT is too big plus Verify() must include DB query to compare stored token and token we get which also means we need to hash JWT and save it in DB.